### ISSUE TITLE
Wall rendering bug with portal at specific position

### ISSUE THREAD

**Issue by Deconimus** [mwpenny/portal64-still-alive#30]
#### Version of the game
 - [latest](https://github.com/mwpenny/portal64-still-alive/commit/e2b3fcdd8d7e79da77d308fa65c56ff1b5a137de)
 - (tested it with previous versions, also happens at e.g. [c90b8ba](https://github.com/mwpenny/portal64-still-alive/commit/c90b8ba42871d0f951ef2ed62e85b96b787791cf))

#### Original Hardware or Emulator?
Emulator. Both Project64 3.0.1 and RMG 0.5.7

#### What is the bug?
Placing portals at specific locations at the very edge of a surface can cause rendering bugs for the surface. It almost looks like the surface is being re-triangulated around the portal with some faces becoming invisible/removed.

#### Steps to reproduce the behavior:
1. In test chamber04
2. Shoot Portal at right wall near the pit, right on the edge of the wall
3. See through the wall

https://github.com/mwpenny/portal64-still-alive/assets/2451901/8f8e02f1-8465-4ecc-845c-46cd95a493f6

**Comment by mwpenny** [comment]
Test chamber 4 is really proving to be a good map for finding issues.

I suspect the culprit here is the hole cutting logic for portal placement. The criteria seems rather specific - I tried in a few other chambers and couldn't reproduce there.

I won't be looking at this one right away given the low severity. In the mean time, feel free. Otherwise, there are some other enhancements I want to do to portal placement and I'll investigate then.

**Comment by Deconimus** [comment]
> Test chamber 4 is really proving to be a good map for finding issues.

True. Here's another similar phenomenon I accidentally stumbled accross. Not sure if it's same issue, but looks like the inverse of the problem.


https://github.com/mwpenny/portal64-still-alive/assets/2451901/5cc3dc98-2e26-41f0-9a23-8e1c9f920926

**Comment by mwpenny** [comment]
Yes, looks related. It's also a good idea to try on a more accurate emulator like Ares, but likely these are both real.
